processing/src/main/java/io/druid/collections/spatial/ImmutableRTree.java

```diff
@@ -75,7 +74,7 @@ public static ImmutableRTree newImmutableFromMutable(RTree rTree)
       return empty();
     }
 
-    ByteBuffer buffer = ByteBuffer.wrap(new byte[calcNumBytes(rTree)]);
+    ByteBuffer buffer = ByteBuffer.allocate(calcNumBytes(rTree));
```

Comment:
Maybe using `ByteBuffer.allocateDirect` will get better performance.

```java
ByteBuffer buffer = ByteBuffer.allocateDirect(calcNumBytes(rTree));
```

Reply:
Direct ByteBuffers are not the thing that "magically makes everything faster". There are two main reasons to use direct ByteBuffers:
 - Reduce the size of Java Heap, (ordinary BBs are backed with `byte[]` arrays which are in the heap), to reduce GC pause times. Yes, even heap populated with arrays of primitives could affect GC pause times negatively. However, this comes with many downsides:
   - Native allocations are slower
   - Native allocations contribute to potentially unrecoverable fragmentation. On the contrary, Java Heap fragmentation is cleaned up, sooner or later.
   - You need ensure timely freeing of direct BBs, preferably with try-with-resources, not relying on reference cleaning

 - Avoid extra data copies when going native, particularly when a ByteBuffer participate network or disk IO calls (ByteChannel.read() / write())

Neither of this directly applies to `ImmutableRTree`.